core/src/main/java/com/airbnb/aerosolve/core/models/MlpModel.java

```diff
@@ -0,0 +1,326 @@
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.io.Serializable;
+import java.util.*;
+
+import com.airbnb.aerosolve.core.DebugScoreRecord;
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.airbnb.aerosolve.core.ModelHeader;
+import com.airbnb.aerosolve.core.ModelRecord;
+import com.airbnb.aerosolve.core.util.Util;
+import com.airbnb.aerosolve.core.util.FloatVector;
+import lombok.Getter;
+import lombok.Setter;
+
+
+/**
+ * Multilayer perceptron (MLP) model https://en.wikipedia.org/wiki/Multilayer_perceptron
+ * The current implementation is for the case where there is only one output node.
+ */
+public class MlpModel extends AbstractModel {
+
+  private static final long serialVersionUID = -6870862764598907090L;
+
+  // weights that define the projection from input layer (layer0) to the first hidden layer
+  // or the output layer
+  @Getter
+  @Setter
+  private Map<String, Map<String, FloatVector>> inputLayerWeights;
+
+  // if there is hidden layer, this defines the projection
+  // from one hidden layer to the next hidden layer or output layer
+  // map from hidden layer id to hidden layer weights, id starts from 0
+  @Getter
+  @Setter
+  private Map<Integer, ArrayList<FloatVector>> hiddenLayerWeights;
+
+  // map from layer Id to bias applied on each node in the layer
+  @Getter
+  @Setter
+  private Map<Integer, FloatVector> bias;
+
+  @Getter
+  @Setter
+  private String activationFunction;
+
+  // number of layers (excluding input layer and output layer)
+  @Getter
+  @Setter
+  private int numHiddenLayers;
+
+  // number of nodes for each hidden layer and output layer (does not include input layer)
+  @Getter
+  @Setter
+  private Map<Integer, Integer> layerNodeNumber;
+
+  @Getter
+  @Setter
+  private Map<Integer, FloatVector> layerActivations;
+
+  public MlpModel() {
+    layerNodeNumber = new HashMap<>();
+    inputLayerWeights = new HashMap<>();
+    hiddenLayerWeights = new HashMap<>();
+    layerActivations = new HashMap<>();
+    bias = new HashMap<>();
+  }
+
+  public MlpModel(String activation, ArrayList<Integer> nodeNumbers) {
+    // n is the number of hidden layers (does not count input and output layer)
+    // activation specifies activation function
+    // nodeNumbers: specifies number of nodes in each hidden layer
+    numHiddenLayers = nodeNumbers.size();
+    activationFunction = activation;
+    layerNodeNumber = new HashMap<>();
+    inputLayerWeights = new HashMap<>();
+    hiddenLayerWeights = new HashMap<>();
+    // bias including the bias added at the output layer
+    bias = new HashMap<>();
+    layerActivations = new HashMap<>();
+
+    for (int i = 0; i < numHiddenLayers; i++) {
+      int nodeNum = nodeNumbers.get(i);
+      layerNodeNumber.put(i, nodeNum);
+      hiddenLayerWeights.put(i, new ArrayList<>(nodeNum));
+      bias.put(i, new FloatVector(nodeNum));
+      layerActivations.put(i, new FloatVector(nodeNum));
+    }
+    // output layer has one node
```

Comment:
yes, this is just for binary classification and regression. I may add the multi-class case later